src/engine/TextIndexScanForWord.cpp

```diff
@@ -56,32 +41,24 @@ VariableToColumnMap WordIndexScan::computeVariableToColumnMap() const {
     addDefinedVar(
         cvar_.getMatchingWordVariable(word_.substr(0, word_.size() - 1)));
   }
-  addDefinedVar(cvar_.getScoreVariable());
-  for (const auto& var : variables_) {
```

Comment:
I think the variant that reproduces the previous behavior best would be to only
read the score variable for exactly one of the `text index scans`. So therefore you would need an additional argument to the constructor whether we also want the scores. I understand that adding them to all the results doesn't work or at least not efficiently.